src/Data/Managers/V2/Notification/GetNotificationsQL/useGetNotificationQL.tsx

```diff
@@ -0,0 +1,99 @@
+import { useQuery, UseQueryResult } from "@tanstack/react-query";
+import { gql } from "graphql-request";
+import { Notification } from "@/Slices/Notification/Core/Domain";
+import { CustomError } from "../../helpers";
+import { useGraphQLRequest } from "../../helpers/useGraphQL";
+
+interface Props {
+  envID: string;
+  cleared: boolean;
+  orderBy: string;
+}
+
+/**
+ * Response type for the notifications query through GraphQL.
+ *
+ * @property {Object} data - The data object containing the notifications.
+ * @property {Array} errors - The errors array containing any errors that occurred.
+ * @property {Object} extensions - The extensions object containing any additional data.
+ */
+export interface NotificationQLResponse {
+  data: {
+    notifications: {
+      edges: {
+        node: PartialNotification;
+      }[];
+    };
+  };
+  errors: string[] | null;
+  extensions: Record<string, unknown>;
+}
+
+/**
+ * Return Signature of the useGetInstance React Query
+ */
+interface GetNotifications {
+  useContinuous: () => UseQueryResult<NotificationQueryResponse, CustomError>;
+}
+
+/**
+ * Partial Notification type that represents the Notification type but with only the title, severity, and read properties.
+ */
+export type PartialNotification = Pick<Notification, "title" | "severity" | "read">;
+
+/**
+ * Response of the useGetNotificationQL hook after it has been processed.
+ *
+ * @property {PartialNotification[]} data - The data object containing the notifications.
+ * @property {string[] | null} errors - The errors array containing any errors that occurred.
+ * @property {Record<string, unknown>} extensions - The extensions object containing any additional data.
+ */
+export interface NotificationQueryResponse {
+  notifications: PartialNotification[];
+  errors: string[] | null;
+  extensions: Record<string, unknown>;
+}
+
+/**
+ * React Query hook for fetching notifications using GraphQL.
+ *
+ * @param {Props} props - The props object containing environment ID, cleared status, and order by.
+ * @returns GetNotifications A query result containing notifications data or an error.
+ */
+export const useGetNotificationQL = ({ envID, cleared, orderBy }: Props): GetNotifications => {
```

Comment:
I wouldn't use QL as an identifier for the name of the query. This is the limited notification list we need for the app-shell. It would make sense to me to use a name that would reflect that instead. Especially considering you're already including extensions in the response, making it more than about Notifications.

Also be careful about plural/singular wording. `Notifications != Notification`